src/ray/object_manager/plasma/store.h

```diff
@@ -273,6 +284,19 @@ class PlasmaStore {
   /// complete.
   ray::SpillObjectsCallback spill_objects_callback_;
 
+  /// Queue of object creation requests to respond to. Requests will be placed
+  /// on this queue if the object store does not have enough room at the time
+  /// that the client made the creation request, but space may be made through
+  /// object spilling. Once the raylet notifies us that objects have been
+  /// spilled, we will attempt to process these requests again and respond to
+  /// the client if successful or out of memory. If more objects must be
+  /// spilled, the request will be replaced at the head of the queue.
+  /// TODO(swang): We should also queue objects here even if there is no room
```

Comment:
Maybe it's not clear from the comment, but there are now two types of OutOfMemory errors, transient (but we expect there to be space once spilling is done) or definite (unless some objects get released, there's definitely no space). Right now we only queue requests for the former.

Reply:
Can we remove the TODO then? I think it doesn't make sense to queue the latter (should just raise OOM if only mmaped objects are in the store).

Reply:
Actually I think it does make sense to eventually queue the latter too. Right now we have to retry from the client on OOM, but that can add unnecessary delays in cases where other objects will actually get released soon. If we queue on the plasma store (with a timeout), we can create the object as soon as there's enough space. Also we could potentially reduce false positives if the plasma store makes the decision instead of the client.

Reply:
Right yeah, I think eventually we want all decision making to be centralized in this queue (including trigger of global gc, etc.).

Reply:
The latter is possible after we make create asynchronous right?